homeassistant/components/rituals_perfume_genie/__init__.py

```diff
@@ -30,27 +31,30 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
 
     try:
         account_devices = await account.get_devices()
-    except ClientConnectorError as ex:
-        raise ConfigEntryNotReady from ex
+    except aiohttp.ClientError as err:
+        raise ConfigEntryNotReady from err
 
-    hublots = []
     devices = {}
     for device in account_devices:
         hublot = device.data[HUB][HUBLOT]
-        hublots.append(hublot)
         devices[hublot] = device
 
     hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {
         COORDINATORS: {},
         DEVICES: devices,
     }
 
-    for hublot in hublots:
-        device = hass.data[DOMAIN][entry.entry_id][DEVICES][hublot]
+    for hublot, device in devices.items():
```

Comment:
Another optimization would be to merge this iteration with the one above on line 38.